packages/components/radio/src/radio.js

```diff
@@ -0,0 +1,63 @@
+import { TSElement, unsafeCSS, html, customElementDefineHelper } from '@tradeshift/elements';
+import css from './radio.css';
+import { customEventNames } from './utils/constants';
+
+customElementDefineHelper(
+	'ts-radio',
+	class extends TSElement {
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(css)];
+		}
+
+		static get properties() {
+			return {
+				value: { type: String, reflect: true },
+				label: { type: String, reflect: true },
+				checked: { type: Boolean, reflect: true },
+				disabled: { type: Boolean, reflect: true },
+				dir: { type: String, reflect: true }
+			};
+		}
+
+		handleClick(e) {
+			e.preventDefault();
+			e.stopPropagation();
+			this.dispatchEvent(new Event(customEventNames.RADIO_CLICK, { bubbles: true, composed: true }));
+		}
+
+		get direction() {
+			return this.dir || this.bodyDir;
+		}
+
+		handleFocus() {
+			this.dispatchEvent(new Event(customEventNames.RADIO_FOCUS, { bubbles: true, composed: true }));
+		}
+
+		handleBlur() {
+			this.dispatchEvent(new Event(customEventNames.RADIO_BLUR, { bubbles: true, composed: true }));
+		}
+
+		render() {
+			return html`
+				<label
+					tabindex="-1"
+					dir="${this.direction}"
+					@click="${this.handleClick}"
+					@focus="${this.handleFocus}"
+					@blur="${this.handleBlur}"
+				>
+					${this.label}
+					<input
+						tabindex="-1"
+						type="radio"
+						role="radio"
+						.value="${this.value}"
+						?checked="${this.checked}"
+						?disabled="${this.disabled}"
+					/>
+					<span></span>
```

Comment:
Can we use pseudo elements instead? `:after`?

Reply:
Yes, we can. Also, I am not sure if we need to have `<input>` element inside. It has no graphical purpose and placed here only for semantics. I suppose we can reduce radio to a single label element.

Reply:
Fixed

Reply:
>Yes, we can. Also, I am not sure if we need to have <input> element inside. It has no graphical purpose and placed here only for semantics. I am not sure if semantics have any sense inside shadow dom, I suppose we can reduce radio template to a single label element.


Can you please check them in form data? I think we need to keep the inputs anyway for future proofing the input related events!

Reply:
@kirill-kolombet What was the result?

Reply:
I created a simple form to check the behavior of inputs inside radio shadow DOM. If the input element inside, it's content not accessible with the form data

```
	let form = document.querySelector('#radio-form');
	form.addEventListener('submit', (e) => {
		e.preventDefault();
		const form = document.querySelector('form');
		const data = Object.fromEntries(new FormData(form).entries()); //data is empty
	});
```

Currently, the only way to get the value of the radio group is to subscribe to the event: 
```
    this.refs['radio'].addEventListener(
    	'radio-selected', (e) => { this.setState({value: e.detail.value}); }
    );
```

Or directly get value on submit: 
```
    const value = this.refs['radio'].value;
```

If we want ts-radio-group to work with the form data we can:
1. Use hidden <input> element instead of <ts-radio> inside <ts-radio-group> slot. Generate visual representation of those inputs with the <ts-radio> in the shadow DOM, and update slotted inputs state on change
2. Generate hidden <input> with unique id element in the light DOM, next to <ts-radio-group>, like it's done in the weightless components. The simple and effective way.

Reply:
I would like to delay this decision until we'll make more custom input controls and look at what would work best for us.